vac/raw/logos-routing-records.md

```diff
@@ -0,0 +1,158 @@
+and essential information pertaining to those services,
+supported by the peer.
+
+There are at least two reasons why a peer might want to encode service information in its routing records:
+
+1. **To augment `identify` with peer capabilities:**
+The libp2p [`identify` protocol](https://github.com/libp2p/specs/blob/7740c076350b6636b868a9e4a411280eea34d335/identify/README.md) allows peers to exchange critical information,
+such as supported protocols,
+on first connection.
+The routing record (in a signed envelope) can also be exchanged during `identify`.
+However, peers may want to exchange finer-grained information related to supported protocols/services,
+that would otherwise require an application-level negotiation protocol.
+An example would be nodes supporting libp2p [`mix` protocol](https://rfc.vac.dev/vac/raw/mix) also needing to exchange the mix key
+before the service can be used.
+2. **To advertise supported services:**
+If the routing record is used as the discoverable record for a peer
+(as we propose for Logos discovery methods)
+that peer may want to encode a list of supported services
+in its advertised record.
+These services may be (but is not limited to) a list of supported libp2p protocols
+and critical information pertaining to that service (such as the mix key, explained above).
+Discoverers can then filter discovered records for desired capabilities
+based on the encoded service information
+or use it to initiate the service.
+
+## Wire protocol
+
+### Logos Routing Records
+
+Logos Routing Records MUST adhere to the following structure:
+
+```protobuf
+syntax = "proto3";
+
+package logos.peer.pb;
+
+// LogosPeerRecord messages contain information that is useful to share with other peers.
+// Currently, a LogosPeerRecord contains the public listen addresses for a peer
+// and an extensible list of supported services as key-value pairs.
+//
+// LogosPeerRecords are designed to be serialised to bytes and placed inside of
+// SignedEnvelopes before sharing with other peers.
+message LogosPeerRecord {
+
+  // AddressInfo is a wrapper around a binary multiaddr. It is defined as a
+  // separate message to allow us to add per-address metadata in the future.
+  message AddressInfo {
+    bytes multiaddr = 1;
+  }
+
+  // peer_id contains a libp2p peer id in its binary representation.
+  bytes peer_id = 1;
+
+  // seq contains a monotonically-increasing sequence counter to order LogosPeerRecords in time.
+  uint64 seq = 2;
+
+  // addresses is a list of public listen addresses for the peer.
+  repeated AddressInfo addresses = 3;
+
+  message ServiceInfo{
+    string id = 1;
+    optional bytes data = 2;
+  }
+
+   // Extensible list of advertised services
+  repeated ServiceInfo services = 4;
+}
+```
+
+A peer MAY include a list of supported services in the `services` field.
+These services could be libp2p protocols,
+in which case it is RECOMMENDED that the libp2p protocol identifier be used as the `ServiceInfo` `id` field.
```

Comment:
```suggestion
in which case it is MUST that the libp2p protocol identifier be used as the `ServiceInfo` `id` field.
```

The cap. disco. spec and this should be aligned. Maybe just refer to it or vice versa?